src/charm.py

```diff
@@ -95,6 +175,117 @@ def _pebble_layer(self) -> ops.pebble.LayerDict:
             },
         }
 
+    @property
+    def _application_bind_address(self) -> str | None:
+        binding = self.model.get_binding("juju-info")
+        if not binding:
+            return None
+        if not binding.network:
+            return None
+        if not binding.network.bind_address:
+            return None
+        return str(binding.network.bind_address)
+
+    @property
+    def _certificate_requests_table(self) -> List[dict[str, str]] | None:
+        # TODO: Error checking
+        r = requests.get(
+            url=f"https://{self._application_bind_address}:{self.port}/api/v1/certificate_requests",
+            verify=f"{CHARM_PATH}/{CONFIG_MOUNT}/0/ca.pem",
+        )
+        gocert_csrs = r.json()
+        return gocert_csrs
```

Comment:
I'd encourage to include the error checking in the same PR

Reply:
I may just move the entire function to a different branch, because I want to have its tests in the same PR, but if I add those here this PR the size will increase a lot. is that ok?

Reply:
I removed this logic from here and pushed to a [different branch](https://github.com/canonical/gocert-k8s-operator/pull/6).